be/src/storage/vectorized/column_expr_predicate.cpp

```diff
@@ -176,6 +177,70 @@ std::string ColumnExprPredicate::debug_string() const {
     return ss.str();
 }
 
+Status ColumnExprPredicate::try_to_rewrite_for_zone_map_filter(starrocks::ObjectPool* pool,
+                                                               std::vector<const ColumnExprPredicate*>* output) {
+    DCHECK(pool != nullptr);
+    DCHECK(output != nullptr);
+    ExprContext* pred_from_planner = _expr_ctxs[0];
+    Expr* root = pred_from_planner->root();
+
+    std::vector<Expr*> exprs_after_rewrite;
+
+    if (root->op() == TExprOpcode::EQ) {
+        DCHECK(root->get_num_children() == 2);
```

Comment:
Maybe need an explicit check?

Reply:
> Maybe need an explicit check?

what do you mean by `explicit check`? if `root->get_num_children() != 2` , we just return?

Reply:
yes